sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c

```diff
@@ -249,28 +252,12 @@ static ng_pktsnip_t *_pktbuf_duplicate(const ng_pktsnip_t *pkt)
         return NULL;
     }
 
+    DEBUG("pktbuf: copying %u byte from %p to %p\n", (unsigned)pkt->size,
+          pkt->data, res->data);
     memcpy(res->data, pkt->data, pkt->size);
     res->type = pkt->type;
-
-    while (pkt->next) {
-        ng_pktsnip_t *hdr = NULL;
-
-        pkt = pkt->next;
-        hdr = _pktbuf_add_unsafe(res, pkt->data, pkt->size, pkt->type);
-
-        if (hdr == NULL) {
-            do {
-                ng_pktsnip_t *next = res->next;
-
-                _pktbuf_internal_free(res->data);
-                _pktbuf_internal_free(res);
-
-                res = next;
-            } while (res);
-
-            return NULL;
-        }
-    }
```

Comment:
For clarification: the current code duplicates the whole packet, while this PR would result in duplicating only the current sniplet itself, right?